sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,412 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import logging
+
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+if daal_check_version((2024, "P", 600)):
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.linear_model import Ridge as sklearn_Ridge
+    from sklearn.metrics import r2_score
+    from sklearn.utils.validation import check_is_fitted, check_X_y
+
+    from daal4py.sklearn.linear_model._ridge import _fit_ridge as daal4py_fit_ridge
+
+    if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+        from sklearn.linear_model._base import _deprecate_normalize
+
+    from onedal.linear_model import Ridge as onedal_Ridge
+    from onedal.utils import _num_features, _num_samples
+
+    from ..._device_offload import dispatch, wrap_output_data
+    from ..._utils import PatchingConditionsChain
+    from ...utils import get_namespace
+    from ...utils.validation import _assert_all_finite
+
+    def _is_numeric_scalar(value):
+        """
+        Determines if the provided value is an instance of either int or float.
+
+        Args:
+        value: The value to be checked.
+
+        Returns:
+        bool: True if the value is either an int or a float, False otherwise.
+        """
+        return isinstance(value, (int, float))
+
+    def _test_type_and_finiteness(X_in):
+        """
+        Checks if the input is of a supported type and is finite.
+
+        Args:
+        X_in: The input to be checked.
+
+        Returns:
+        bool: True if the input is of a supported type and is finite, False otherwise.
+        """
+
+        xp, _ = get_namespace(X_in)
+        X = xp.asarray(X_in)
+
+        if np.iscomplexobj(X):
+            return False
+
+        try:
+            _assert_all_finite(X)
+        except BaseException:
+            return False
+
+        return True
```

Comment:
scikit-learn should have functions that cover this (ie _validate_params, check_xy etc.) - @icfaust can you confirm this?

Reply:
This is a carry-over from [LinearRegression](https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/linear_model/linear.py#L141) (and has found its way into [LogisticRegression](https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/linear_model/logistic_regression.py#L180) too). Ideally we would remove this bad design pattern from all three.  The finite assert occurs in sklearn [LinearRegression](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_base.py#L600), meaning it is duplicated here unnecessarily and is causing an unnecessary performance loss in LinearRegression and LogisticRegression as well (O(n)).  This whole function can be converted into the use of a single np.iscomplexobj call, since that will work with values, lists, arrays and array like objects (no conversion to numpy necessary).  I would remove it here, and we make another PR for LogisticRegression and LinearRegression separately.  Thanks @ethanglaser for making me dive in, I should have done this on the LogisticRegression out of preview PR.

Reply:
Now implemented in #1923